File: 1816-truncate-sentence/1816-truncate-sentence.kt
class Solution {
fun truncateSentence(s: String, k: Int): String {
    var result = ""
    var currentWords = 0

    for (i in s.indices) {
        if (s[i] == ' ') {
            currentWords += 1

            if (currentWords == k) return result
        }

        result += s[i]
    }

    return result
}

}

File: 1816-truncate-sentence/1816-truncate-sentence.rs
impl Solution {
    pub fn truncate_sentence(s: String, k: i32) -> String {
        let splited = s.split(" ");
        let mut result = String::new();
        let mut counter = 0;
        for word in splited {
            if counter == k {
                break;
            }
            result.push_str(word);
            counter += 1;

            result.push_str(" ");
        }

        result.remove(result.len() - 1);

        return result;
    }
}
